### PR TITLE
Implement correct Authentication

### LINKED ISSUE #38: As a user, I should be able to create an account using my email address

### LINKED ISSUE #39: As an android user, I should be able to authenticate with my Google Account

### PR DESCRIPTION
**Pull request by hamzaremmal**
This PR adds Authentication to Versus, it also starts the implementation of Google Authentication. 

From a dev standpoint, this PR does:
- Enable View Binding
- add as a dependency : 'org.mockito:mockito-inline:5.2.0', 'org.mockito:mockito-android:5.2.0', 'org.mockito:mockito-core:5.2.0' & 'com.google.android.gms:play-services-auth:20.4.1' dependencies

Tests are not the best, they will be changed and fixed in the next sprint.

### CODE REVIEW

File: app/src/main/java/com/github/versus/EntryActivity.java [hunk 1]
```diff
@@ -61,28 +61,12 @@ private void yieldActivity(@Nullable FirebaseUser user){
...
-
-    /**
-     * ???
-     * @param auth
-     */
-    private void handleIdTokenChanged(FirebaseAuth auth){
-        // TODO HR : To protect the App from LogIn issues, add listeners
-        //  so we can handle such errors
+    private void initAuthentication() {
+        // HR : For testing purposes, we check if the intent contains an authenticator
```
Comment (ktotam1) [inline app/src/main/java/com/github/versus/EntryActivity.java:65]: figure out how to refactor so that testing can be done entirely from tests

**Review by ktotam1 — APPROVED** [review]
only allow android studio to clean files you are working on because it is hard to check so many files that is just refactoring.

File: app/src/main/java/com/github/versus/EntryActivity.java (hunk 1 above)
Comment (hamzaremmal) [reply app/src/main/java/com/github/versus/EntryActivity.java:65]: Didn't have much time to learn Firebase Emulator, I'm planning on doing that next sprint :-)

**Review by DerouichAbdessalam — APPROVED** [review]
good tests

**Review by DerouichAbdessalam — APPROVED** [review]
Nice auth implementation, could use more coverage but tests are quite comprehensive for now,

**Comment by ziazi** [comment]
Nice modularisation and use of mockito framework.

**Pull request merged**